src/main/java/de/tum/in/www1/artemis/service/TextAssessmentService.java

```diff
@@ -78,6 +78,9 @@ public Result saveAssessment(Long resultId, List<Feedback> textAssessment, TextE
         }
 
         result.setAssessmentType(AssessmentType.MANUAL);
+        if (textAssessment.stream().filter(feedback -> feedback.getType() == FeedbackType.AUTOMATIC || feedback.getType() == FeedbackType.AUTOMATIC_ADAPTED).count() > 0) {
```

Comment:
I wonder if we should now move setting the assessmentType to e.g. a class method, just to clean up the code here a bit and make it reusable.
So instead of
`result.setAssessmentType(AssessmentType.MANUAL)`
it would be:
`result.setAssessmentType();`
Where the method determines based on the result's feedback items what kind of assessment it has.

Reply:
Good idea, or at least offer this method as an option and call it `result.autoDetermineAssessmentType()`.

Reply:
> I wonder if we should now move setting the assessmentType to e.g. a class method, just to clean up the code here a bit and make it reusable.
> So instead of
> `result.setAssessmentType(AssessmentType.MANUAL)`
> it would be:
> `result.setAssessmentType();`
> Where the method determines based on the result's feedback items what kind of assessment it has.

Good idea, or at least offer this method as an option and call it `result.autoDetermineAssessmentType().`

Reply:
That is a good Idea. I added it